webClient/src/app/connection-panel/connection-panel.component.ts

```diff
@@ -0,0 +1,80 @@
+import { Component, EventEmitter, Output, Inject} from '@angular/core';
+import { Connection } from '../Connection';
+import { FTAWebsocketService } from '../services/FTAWebsocket.service';
+
+import { SelectItem } from 'primeng/api';
+import { Message } from 'primeng/components/common/api';
+import { Angular2InjectionTokens } from 'pluginlib/inject-resources';
+
+@Component({
+  selector: 'app-connection-panel',
+  templateUrl: './connection-panel.component.html',
+  styleUrls: [
+  '../../../node_modules/carbon-components/css/carbon-components.min.css',
+  './connection-panel.component.css'
+  ],
+  providers: [FTAWebsocketService]
+})
+export class ConnectionPanelComponent {
+
+  @Output() submitted = new EventEmitter<Connection>();
+
+  title = 'app';
+  ftaServiceUrl: string;
+
+  protocols: SelectItem[];
+  selectedProtocol: SelectItem;
+
+  address = '';
+  username = '';
+  password = '';
+
+  remoteConnection: Connection;
+  connections: Connection[];
+
+  connectionMessages: Message[];
+
+  constructor(@Inject(Angular2InjectionTokens.PLUGIN_DEFINITION) private pluginDefinition: ZLUX.ContainerPluginDefinition) {
+    this.ftaServiceUrl = (window as any).ZoweZLUX.uriBroker.pluginWSUri(this.pluginDefinition.getBasePlugin(), 'fs', '');
```

Comment:
I think you can actually avoid (window as any) and just use ZoweZLUX, implicitly, if that works better, because it will look better.